src/Tools/CodeGenerator/Plugins/OnnxRuntimePlugin.py

```diff
@@ -372,6 +373,34 @@ def CreateOutputStatement(
             """,
         ).format(index, output_name, output_documentation or "No information is available", output_type)
 
+    # ----------------------------------------------------------------------
+    def CreateTypeInferenceConstraints(output_type_mappings):
+        code = []
+        mapping_length = len(output_type_mappings)
+        constraint_format = "input_elem_type == ONNX_NAMESPACE::TensorProto_DataType_{input_type_upper}"
+
+        for index, (output_type, input_types) in enumerate(output_type_mappings.items()):
+            constraints = []
+            for input_type in input_types:
+                constraints.append(constraint_format.format(
+                    input_type_upper = re.sub(r'^std::|_t$', '', input_type).upper()

```

Comment:
Creating a regex is relatively expensive - it would be better to create it outside of the loop and then use it here.

Reply:
I didn't do that because I read this in the documentation.
```
Note The compiled versions of the most recent patterns passed to re.compile() and the module-level matching functions are cached, so programs that use only a few regular expressions at a time needn’t worry about compiling regular expressions.
```
I am using `sub` not `match`, so will it still get cached in this case?

Reply:
Interesting - I didn't know that. My hunch is that it will cache for both sub and match.